### ISSUE TITLE
A few issues

### ISSUE THREAD

**Issue by rizreflects** [nfm886/skin.cosmic#8]
Great work on this skin much like your previous one. It seems faster than any other skin i've used.. Even the ratings load instantly. 

Some problems I've noticed:

1- Clear Logos don't work well with next up. 

![episodes](https://user-images.githubusercontent.com/76912707/128614823-b571bce3-c0c1-4db2-b26a-1c464423fac2.PNG)

The image above is taken from Serens 'Next Up'. But as you can see theres nothing here which indicates which episode I'm on. The only way to find this out is by looking at the date.

![episodes2](https://user-images.githubusercontent.com/76912707/128614941-11fecc6a-20c5-4eb7-9c3c-02d2f219c367.png)

When the clearlogo isn't there it looks much better in this case. Maybe you could provide an option to remove clearlogos on certain widgets?

![episodes3](https://user-images.githubusercontent.com/76912707/128617087-3940bf7a-7eac-4c3f-8738-4db710d2df5a.png)

FYI this is how serens next up looks using titan bingie mod. I'd say it looks perfect in terms of its layout. If you could implement something like this i'd be eternally grateful. 

2- On the ninth widget the label doesn't display.

![9thlabel](https://user-images.githubusercontent.com/76912707/128617492-ca1c7a6a-722f-474a-a7af-f9d259e2cec6.png)

The first 8 widgets work fine, but from 9 onwards the label doesn't display as shown in the picture. I've only tested this once though so you may wanna double check.

3- Problem with the posters.

This ones a little hard to explain but I'll do my best. Lets say I'm scrolling through a widget. I press right on my remote control. The poster on the far right (which just came into the frame) takes about a second to load. This only occurs when its loading a poster thats not already been cached. 

To repeat this all you need to do is install a video addon (in my case seren), add a list such as 'Trending' as a widget, and slowly scroll right. Keep your eye on the poster on the end and hopefully you'll see what I mean.

4- (Request) Cycle studios, genres and actors.

![studio](https://user-images.githubusercontent.com/76912707/128617706-67fad25b-3faa-40fb-ad8a-bed04d3d6ac1.png)

Right now only studios are shown on the home page. Could you possibly make it cycle with genres and actors like htitan bingie? If not, please consider replacing the studios with genres instead.

5- (Request) 'Add to watchlist' when clicking on info.

If you could add that it would be appreciated. Will save me from opening the context menu each time I want to add a movie to my watchlist.

6- (Request) Option to remove certain online ratings

Such as Trakt and TMDB. Also an option to remove the percentage would be appreciated since its just the Trakt ratings.

Sorry if this is a bit much. I've noticed a few other issues but they could be seen as 'nitpicking' so I decided not to mention them. Also I realise this skin is still a work in progress so I'm not expecting it to be perfect. It's still my new daily skin though. I really appreciate your work.

**Comment by nfm886** [comment]
Thanks for your ideas. Now I'm working on fixing submenu, after it I will add weather/settings widgets and then I will be back to your issues.

**Comment by rizreflects** [comment]
Is it okay if I keep updating this thread with new bugs I find?

Heres another one:

Studios / genres overlapping with plot-

![overlap](https://user-images.githubusercontent.com/76912707/128743277-49f08b76-2e7a-4f78-bdd5-30afc55314f3.png)

As you can see here there is considerable overlap in some cases.

![nooverlap](https://user-images.githubusercontent.com/76912707/128743389-ad8bd43a-42bf-4a4d-94a2-7238771f79f4.png)

If you can limit the genres / studios to only use 2 lines (as shown in the picture) this will solve the problem. You could then set it to autoscroll like the plot.

**Comment by nfm886** [comment]
> Is it okay if I keep updating this thread with new bugs I find?

I think it's better to open new issue because this way I can sort out some stuff.

**Comment by nfm886** [comment]
I will fix it, btw :P

**Comment by rizreflects** [comment]
Okay from now on I'll open a new thread each time 👍

**Comment by nfm886** [comment]
Hey, this one: 

> 5- (Request) 'Add to watchlist' when clicking on info.
> 
> If you could add that it would be appreciated. Will save me from opening the context menu each time I want to add a movie to my watchlist.

Do you mean trakt watchlist? I just implemented "Manage" button in DialogVideoInfo so you can do some stuff there, including Trakt management.

**Comment by rizreflects** [comment]
That would do the job I suppose.. I was thinking more of an actual button that says 'watchlist' like how it is in arctic horizon.. also, yes I was referring to trakt watchlist.

**Comment by nfm886** [comment]
> That would do the job I suppose.. I was thinking more of an actual button that says 'watchlist' like how it is in arctic horizon.. also, yes I was referring to trakt watchlist.

Just added Watchlist button in "Search" button's place. And moved "Search" button to Manage.

**Comment by nfm886** [comment]
I did not push update yet.
If you want to try those changes, you may need to install it directly from GitHub as a zip file.
Please download this zip: https://github.com/nfm886/skin.cosmic/archive/refs/heads/master.zip
Then go to settings > Add-ons > Install from zip file

Keep in mind, you will lose auto updates after it. To bring auto updates back, you have to update skin from heppen's repository.

**Comment by rizreflects** [comment]
Ah okay no worries then, I'll wait till you push the update through :-)

**Comment by nfm886** [comment]
> 4- (Request) Cycle studios, genres and actors.
> 
> ![studio](https://user-images.githubusercontent.com/76912707/128617706-67fad25b-3faa-40fb-ad8a-bed04d3d6ac1.png)
> 
> Right now only studios are shown on the home page. Could you possibly make it cycle with genres and actors like htitan bingie? If not, please consider replacing the studios with genres instead.

Just made them Cycle: https://streamable.com/i4k5lp

**Comment by rizreflects** [comment]
You work incredibly fast. Also that looks fantastic.

**Comment by nfm886** [comment]
> 3- Problem with the posters.
> 
> This ones a little hard to explain but I'll do my best. Lets say I'm scrolling through a widget. I press right on my remote control. The poster on the far right (which just came into the frame) takes about a second to load. This only occurs when its loading a poster thats not already been cached.
> 
> To repeat this all you need to do is install a video addon (in my case seren), add a list such as 'Trending' as a widget, and slowly scroll right. Keep your eye on the poster on the end and hopefully you'll see what I mean.

Yeah, I see. But is it a problem? :P
I don't know if I can do much with it. If posters are not cached, the load time is longer... What can I do with it?
It happens only once so I don't know if it's worth to trying fix it.

**Comment by rizreflects** [comment]
> > 3- Problem with the posters.
> > 
> > This ones a little hard to explain but I'll do my best. Lets say I'm scrolling through a widget. I press right on my remote control. The poster on the far right (which just came into the frame) takes about a second to load. This only occurs when its loading a poster thats not already been cached.
> > 
> > To repeat this all you need to do is install a video addon (in my case seren), add a list such as 'Trending' as a widget, and slowly scroll right. Keep your eye on the poster on the end and hopefully you'll see what I mean.
> 
> Yeah, I see. But is it a problem? :P
> I don't know if I can do much with it. If posters are not cached, the load time is longer... What can I do with it?
> It happens only once so I don't know if it's worth to trying fix it.

I understand. I only mentioned it because I never noticed that problem when using other skins like arctic horizon or titan bingie. Do whatever you feel is best in that case.

**Comment by rizreflects** [comment]
Now that you mention it removing the background might be a good idea.. maybe test it out and we can see how it looks?

**Comment by rizreflects** [comment]
> > 4- (Request) Cycle studios, genres and actors.
> > 
> > ![studio](https://user-images.githubusercontent.com/76912707/128617706-67fad25b-3faa-40fb-ad8a-bed04d3d6ac1.png)
> > 
> > Right now only studios are shown on the home page. Could you possibly make it cycle with genres and actors like htitan bingie? If not, please consider replacing the studios with genres instead.
> 
> Just made them Cycle: https://streamable.com/i4k5lp

I just tested out the new update.. you made a lot of improvements, thank you for that.. however I've noticed a slight problem with the cycling. it seems to only cycle when you stay on the same poster.. But when you go onto the next poster It always reverts back to genres. Was this intentional?

Ideally, when the switch happens, it would be better if it stayed on the screen for about 20 seconds (even when scrolling through posters) before switching again. If you want to flesh it out even more, you could add actors and directors aswell.. that's how Netflix and titan bingie do it as far as I'm aware.

**Comment by nfm886** [comment]
> it seems to only cycle when you stay on the same poster.. But when you go onto the next poster It always reverts back to genres. Was this intentional?

Yes. It works like that because not every movie has studio information, for example - Netflix add-on. There is only Genre so if you want to keep that information always on the screen, it must works like that.

> you could add actors and directors aswell.. that's how Netflix and titan bingie do it as far as I'm aware.
I will see what I can do.

> titan bingie

Sorry but it's not Titan Bingie. I know people like some features from there, I understand it. But if you want use Titan Bingie, then go with it. I'm trying to do something different.

**Comment by rizreflects** [comment]
I see.. makes sense. Also I much prefer your skin over titan bingie.. I was just tryna give you some ideas. Sorry if you took offense to that.

**Comment by nfm886** [comment]
> I see.. makes sense. Also I much prefer your skin over titan bingie.. I was just tryna give you some ideas. Sorry if you took offense to that.

Easy, no offence. Just simply reminder ;)
I appreciate every idea.

**Comment by nfm886** [comment]
> The image above is taken from Serens 'Next Up'. But as you can see theres nothing here which indicates which episode I'm on. The only way to find this out is by looking at the date.

Hey,
Could you please show me how other skins work with it? Especially Arctic Horizon/AZ2. I don't use Seren and add-ons like that so can't check...
However, from what I see in Netflix widget, there is no info either and for local files you can see that info in "Info Line"

**Comment by rizreflects** [comment]
Sorry my friend I just woke up 😅 

Give me half an hour.. I'm going to install those skins now and send u a screenshot

**Comment by nfm886** [comment]
> Sorry my friend I just woke up 
> 
> Give me half an hour.. I'm going to install those skins now and send u a screenshot

OK :)
I just did not see it in AH and AZ2 so need to see how it should works :P

**Comment by rizreflects** [comment]
So in Arctic Zephyr Reloaded the episode number is not present when in netflix view (as shown below):

![arctic zephyr next up netflix](https://user-images.githubusercontent.com/76912707/129198881-c5f56bb2-b80b-4710-8213-2368abb2686c.png)

However in it's vertical view it looks okay.. But that's probably because it shows the label rather than clearart:

![arctiz zephyr vertical widgets](https://user-images.githubusercontent.com/76912707/129199151-7104d75e-0d42-4c13-a489-f599d67f3856.png)

Arctic Horizon handles this the best:

![arctic horizon next up](https://user-images.githubusercontent.com/76912707/129199749-3ccb5502-618d-48fd-98e4-c356004ef51c.png)

P.S. Arctic Zephyr 2 Resurrection has the same problems as AZ Reloaded. Not present in Flix View but present when the label is shown rather than the clearlogo.

**Comment by nfm886** [comment]
Oh, I see. So you mean that "Season 2 | Episode 2" info?
I must see how AH has it implemented. I tried to look into it today's morning but could not figure it out. Now I think we can do this :P

**Comment by rizreflects** [comment]
> Oh, I see. So you mean that "Season 2 | Episode 2" info?

Yes this is exactly what I mean.

> I must see how AH has it implemented. I tried to look into it today's morning but could not figure it out. Now I think we can do this :P

That's great to hear. I'm sure if anyone can figure this out you're the person for the job.

**Comment by nfm886** [comment]
Oh wait... So it should indicate season and episode of selected (onfocus) tv show right?

**Comment by nfm886** [comment]
It's seems I looked for wrong direction ^^

**Comment by rizreflects** [comment]
> Oh wait... So it should indicate season and episode of selected (onfocus) tv show right? 

Yes that's correct. When using seren, you will see the season + episode number when focusing on a particular episode (next up or continue watching.) However, in other widgets (for example trending / most popular) you'll only see how many seasons the show has.

**Comment by rizreflects** [comment]
> 
> 
> > > Oh wait... So it should indicate season and episode of selected (onfocus) tv show right?
> > 
> > 
> > Yes that's correct. When using seren, you will see the season + episode number when focusing on a particular episode (next up or continue watching.) However, in other widgets (for example trending / most popular) you'll only see how many seasons the show has.
> 
> Ok, got it working: https://streamable.com/j99uyz
> Something like this? Now just do some tweaks and it will be ready to push.
> 
> #edit
> It did not load it first time when hovering over Loki tv show because one value wasn't set. Had to move around to update it ;)

Brilliant job. That looks perfect. I was going to ask you this at a later time but I may aswell ask now.. Would you consider adding an option to remove where it shows the end time? IMO it's not needed.. but of course if this is how you intended to design it then leave it as it is.

**Comment by nfm886** [comment]
> Brilliant job. That looks perfect.

Awesome. So we can close this ticket I think. Most of things are solved already ;) If you consider to report any bugs or feature request please do this! I appreciate every feedback :)

> Would you consider adding an option to remove where it shows the end time? IMO it's not needed.. but of course if this is how you intended to design it then leave it as it is.

I'm using it ^^ Probably I will add options to customize Info Line but it's not that important for now ;)

**Comment by rizreflects** [comment]
You solved them incredibly fast aswell. Thank you for all your work. I'll keep posting any bugs I find in the future and help in any way i can.

**Comment by rizreflects** [comment]
FYI the episodes / season numbers are not appearing on most widgets.
See here:
https://streamable.com/cpou5d

**Comment by nfm886** [comment]
> FYI the episodes / season numbers are not appearing on most widgets.
> See here:
> https://streamable.com/cpou5d

Oh right. Gonna fix it right now. But fix will be available with next update I think. 3 updates on a day is a bit too much ^^

**Comment by nfm886** [comment]
Maybe you would like to test fixed version? Then we can be sure it will be fully fixed in next update.

**Comment by rizreflects** [comment]
> Maybe you would like to test fixed version? Then we can be sure it will be fully fixed in next update.

I'd be happy to test it. Shall I download the zip directly from GitHub?

**Comment by nfm886** [comment]
> > Maybe you would like to test fixed version? Then we can be sure it will be fully fixed in next update.
> 
> I'd be happy to test it. Shall I download the zip directly from GitHub?

Yes :)

Please download this zip file: https://github.com/nfm886/skin.cosmic/archive/refs/heads/master.zip
After it, when we will fix it, you can back to version from repository to get regular updates, so don't worry :)

**Comment by nfm886** [comment]
> > > > Maybe you would like to test fixed version? Then we can be sure it will be fully fixed in next update.
> > > 
> > > 
> > > I'd be happy to test it. Shall I download the zip directly from GitHub?
> > 
> > 
> > Yes :)
> > Please download this zip file: https://github.com/nfm886/skin.cosmic/archive/refs/heads/master.zip
> > After it, when we will fix it, you can back to version from repository to get regular updates, so don't worry :)
> 
> I just tested this now using the zip file you provided. I haven't noticed any difference.. seems exactly the same as before.

Ok. Let me think. I will be back with fixed version. I hope :D

**Comment by nfm886** [comment]
> > > > Maybe you would like to test fixed version? Then we can be sure it will be fully fixed in next update.
> > > 
> > > 
> > > I'd be happy to test it. Shall I download the zip directly from GitHub?
> > 
> > 
> > Yes :)
> > Please download this zip file: https://github.com/nfm886/skin.cosmic/archive/refs/heads/master.zip
> > After it, when we will fix it, you can back to version from repository to get regular updates, so don't worry :)
> 
> I just tested this now using the zip file you provided. I haven't noticed any difference.. seems exactly the same as before.

Please re-download this zip file: https://github.com/nfm886/skin.cosmic/archive/refs/heads/master.zip and test again.
Waiting for your feedback!

**Comment by rizreflects** [comment]
It's still not working for me.. exactly the same as before.

**Comment by nfm886** [comment]
> It's still not working for me.. exactly the same as before.

Ehh... Ok. I will back to it tomorrow. I'm done for today ^^

**Comment by rizreflects** [comment]
Update: I tested the seasons / episodes on coreELEC and it seemed to be working perfectly.. even through the normal repo. I'm thinking the problem has something to do with windows which is what I was testing it on before. I'll do a fresh install tomorrow before testing again.

**Comment by nfm886** [comment]
> Update: I tested the seasons / episodes on coreELEC and it seemed to be working perfectly.. even through the normal repo. I'm thinking the problem has something to do with windows which is what I was testing it on before. I'll do a fresh install tomorrow before testing again.

Ok. Waiting for your feedback!

**Comment by rizreflects** [comment]
> > Update: I tested the seasons / episodes on coreELEC and it seemed to be working perfectly.. even through the normal repo. I'm thinking the problem has something to do with windows which is what I was testing it on before. I'll do a fresh install tomorrow before testing again.
> 
> Ok. Waiting for your feedback!

On windows it's the same problem even with a fresh install.. very strange.

**Comment by rizreflects** [comment]
On coreleec (Beelink GT King Pro & Odroid n2+) it works perfectly. I have yet to test it on my Nvidia shield which I'll do at some point today.

**Comment by nfm886** [comment]
Yeah, that's strange... For now, I don't have an idea what to do. I have to end one task and then can back to it. We will see what we can do.

**Comment by rizreflects** [comment]
I do still feel an option to remove clearlogos on particular widgets could solve the problem.. Take a look at this video for example:

https://streamable.com/m31g2s

**Comment by nfm886** [comment]
> I do still feel an option to remove clearlogos on particular widgets could solve the problem.. Take a look at this video for example:
> 
> https://streamable.com/m31g2s

Yeah, but... we still have to recognize when clearlogo should show and when not. So basically it's the same as adding InfoLine because here and there we have to know when it's a season/episode and when it's movie f.e.

I'm ending current task, so can look into it.

**Comment by rizreflects** [comment]
If you need any further testing (even for issues unrelating to mine) feel free to send me a message.. I'll always be happy to test.

**Comment by nfm886** [comment]
Sure, thanks!
